frontend/public/module/k8s/index.ts

```diff
@@ -147,6 +163,46 @@ export type MachineDeploymentKind = {
   };
 } & K8sResourceKind;
 
+export type MachineConfigKind = {
+  spec: {
+    osImageURL: string;
+    config: any;
+  }
+} & K8sResourceKind;
+
+export enum MachineConfigPoolConditionType {
+  MachineConfigPoolUpdated = 'Updated',
+  MachineConfigPoolUpdating = 'Updating',
+  MachineConfigPoolDegraded = 'Degraded',
+}
+
+export type MachineConfigPoolKind = {
+  spec: {
+    machineConfigSelector?: Selector
+    machineSelector?: Selector
+    paused: boolean;
+    maxUnavailable: number | string;
+  };
+  status: {
+    observedGeneration?: number;
+    configuration:{
+      name: string;
+      source: ObjectReference[];
+    };
+    machineCount: number;
+    updatedMachineCount: number;
+    readyMachineCount: number;
+    unavailableMachineCount: number;
+    conditions: {
```

Comment:
Can we make this a generic conditions type we can reuse for any resource?

Reply:
Something like this:
```ts
export type K8sResourceCondition<T> = {
  type: T;
  status: ConditionStatus;
  lastTransitionTime: string;
  reason: string;
  message: string;
};
export type MachineConfigPoolCondition = K8sResourceCondition<MachineConfigPoolConditionType>;

export type MachineConfigPoolKind = {
  spec: {
    machineConfigSelector?: Selector
    machineSelector?: Selector
    paused: boolean;
    maxUnavailable: number | string;
  };
  status: {
    observedGeneration?: number;
    configuration:{
      name: string;
      source: ObjectReference[];
    };
    machineCount: number;
    updatedMachineCount: number;
    readyMachineCount: number;
    unavailableMachineCount: number;
    conditions: MachineConfigPoolCondition;
  };
} & K8sResourceKind;
```